test/common/event/dispatcher_impl_test.cc

```diff
@@ -90,6 +96,14 @@ class DispatcherImplTest : public testing::Test {
   TimerPtr keepalive_timer_;
 };
 
+// TODO(mergeconflict): We also need integration testing to validate that the expected histograms
+// are written when `enable_dispatcher_stats` is true. See issue #6582.
```

Comment:
I'm planning to implement integration tests in a following PR, once I've dealt with #6582, if that's still ok.